IdentityCore/src/cache/MSIDMacTokenCache.m

```diff
@@ -488,8 +485,8 @@ - (BOOL)setItemImpl:(MSIDCredentialCacheItem *)item
     }
     
     __block NSDictionary *tokens;
-    dispatch_sync(self.synchronizationQueue, ^{
-        tokens = [[self.cache objectForKey:@"tokens"] copy];
+    dispatch_barrier_sync(self.synchronizationQueue, ^{
```

Comment:
I am not sure how these two former dispatch_sync blocks were racing. self.cache seems to be accessed only for read. Am I missing anything?

Reply:
Understood. Either lock (i.e. barrier) or deep copy is required.